File: Arrays/Best Time to Buy and Sell Stock.go
func maxProfit(prices []int) int {
    low := prices[0]
    res := 0

    for i := 0; i < len(prices); i++ {
        low = min(prices[i], low)
        res = max(prices[i] - low, res)
    }
    return res;
}

File: Arrays/Best Time to Buy and Sell Stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut min = prices[0];
        let mut res = 0;

        for num in prices {
            min = min.min(num);
            res = res.max(num - min);
        }

        res
    }
}